org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SnippetCompletionProposal.java

```diff
@@ -433,6 +433,9 @@ private static boolean accept(ICompilationUnit cu, CompletionContext completionC
 				if (acceptClass && node instanceof CompletionOnSingleNameReference) {
 					if (completionContext.getEnclosingElement() instanceof IMethod) {
 						CompilationUnit ast = CoreASTProvider.getInstance().getAST(cu, CoreASTProvider.WAIT_YES, null);
+						if (ast == null) {
+							return true;
```

Comment:
I'm returning `true` here, to be consistent with L440. But TBH, that seems fishy to me. I'm not sure if that original choice is a bug or not.

Reply:
It is a bit buggy already but that's probably from before, so I'll just merge for now. The idea seems to be to restrict when the type declaration snippets (`class`, `record`, `interface`) may appear. You can declare a class within a method declaration provided you're outside of some invocation. That's what it seems to be permitting in that block.